tools/windows-node-installer/pkg/cloudprovider/aws/ec2_test.go

```
@@ -6,14 +6,55 @@ import (
 	"crypto/x509"
 	"encoding/pem"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/openshift/windows-machine-config-bootstrapper/tools/windows-node-installer/pkg/types"
 	"github.com/stretchr/testify/assert"
 	"testing"
 )
 
+type mockEC2Client struct {
+	ec2iface.EC2API
+}
+
+type Instance struct {
+	name         string
+	passwordData string
+	state        int64
+}
+
 var (
-	awsProvider = AwsProvider{}
+	mockClient         = &mockEC2Client{}
+	awsProvider        = AwsProvider{EC2: mockClient}
+	runningInstance    = Instance{name: "RunningInstance", passwordData: "abc123", state: int64(16)}
+	terminatedInstance = Instance{name: "terminatedInstance", state: int64(48)}
 )
 
+func (m *mockEC2Client) GetPasswordData(instanceDataInput *ec2.GetPasswordDataInput) (*ec2.GetPasswordDataOutput, error) {
+	instanceId := *instanceDataInput.InstanceId
+	if instanceId == runningInstance.name {
+		return &ec2.GetPasswordDataOutput{
+			InstanceId:   &runningInstance.name,
+			PasswordData: &runningInstance.passwordData,
+		}, nil
+	}
+	return nil, nil
+}
+
+func (m *mockEC2Client) DescribeInstances(instanceInput *ec2.DescribeInstancesInput) (*ec2.DescribeInstancesOutput, error) {
+	instanceId := *instanceInput.InstanceIds[0]
```

You're only handling the first InstanceId, maybe throw an error if len is > 0?